app/src/main/java/org/jellyfin/androidtv/search/SearchProvider.java

```diff
@@ -54,6 +54,10 @@ private void search(String query, boolean delayed) {
             return;
         }
 
+        // Don't search the same thing twice
+        if (query.equals(lastQuery)) return;
```

Comment:
May want to add a null check for query here

Reply:
There is a check just above this one which uses Utils.isEmpty. That function checks for null values already.